File: aa-integration-tests/tests/fixtures/agents/python/single_agent/langgraph_agent.py
"""Single LangGraph agent fixture — F116 E2E acceptance (AAASM-1513).

Scenario: single_agent / framework: langgraph
"""

from __future__ import annotations

import os
import sys

sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
from _shared import emit, load_config


def run_real(cfg: dict) -> None:
    import asyncio
    from typing import TypedDict

    from langgraph.graph import END, StateGraph

    from agent_assembly import init_assembly

    ctx = init_assembly(
        gateway_url=cfg["gateway_addr"],
        api_key="e2e-test-key",
        agent_id=cfg["agent_id"],
        team_id="f116-e2e",
        mode="sdk-only",
    )
    asyncio.run(ctx.client.register_agent())
    emit({"event": "started", "agent_id": cfg["agent_id"], "framework": "langgraph"})

    # Minimal StateGraph — init_assembly() patches StateGraph.compile() via
    # LangGraphPatch so node wrappers fire governance hooks on every invocation.
    class State(TypedDict):
        task: str
        result: str

    def process_node(state: State) -> dict[str, str]:
        return {"task": state["task"], "result": f"processed: {state['task']}"}

    graph = StateGraph(State)
    graph.add_node("process", process_node)
    graph.set_entry_point("process")
    graph.add_edge("process", END)
    output = graph.compile().invoke({"task": cfg["task"], "result": ""})
    emit({"event": "tool_call", "tool": "langgraph_process_node", "input": cfg["task"]})
    ctx.shutdown()
    emit({"event": "done", "result": output["result"]})


if __name__ == "__main__":
    cfg = load_config()
    if os.environ.get("AA_SELFTEST") == "1":
        emit({"event": "started", "agent_id": cfg["agent_id"], "framework": "langgraph"})
        emit({"event": "tool_call", "tool": "langgraph_mock_tool", "input": ""})
        emit({"event": "done", "result": "selftest-ok"})
        sys.exit(0)
    run_real(cfg)

File: aa-integration-tests/tests/fixtures/agents/typescript/single_agent/langgraph_agent.ts
// F116 ST-B — single-agent LangGraph fixture (AAASM-1514).
//
// Registers one agent, builds a minimal two-node LangGraph (start → echo_node
// → END), invokes it, then shuts down. In selftest mode emits synthetic events.
//
// Invocation:
//   AA_GATEWAY_ADDR=127.0.0.1:PORT AA_AGENT_ID=e2e-lg \
//     pnpm exec tsx single_agent/langgraph_agent.ts
//
//   AA_SELFTEST=1 AA_GATEWAY_ADDR=dummy pnpm exec tsx single_agent/langgraph_agent.ts

import { loadConfig, emit, type AgentConfig } from "../_shared.js";
import { initAssembly } from "@agent-assembly/sdk";
import { StateGraph, Annotation } from "@langchain/langgraph";

const StateAnnotation = Annotation.Root({
  task: Annotation<string>(),
  result: Annotation<string>(),
});

async function runReal(cfg: AgentConfig): Promise<void> {
  const ctx = await initAssembly({
    gatewayUrl: `http://${cfg.gatewayAddr}`,
    apiKey: "e2e-test-key",
    agentId: cfg.agentId,
    teamId: "f116-e2e",
    mode: "sdk-only",
  });

  emit({ event: "started", agent_id: cfg.agentId });

  const graph = new StateGraph(StateAnnotation)
    .addNode("echo_node", async (state) => ({
      result: `echo: ${state.task}`,
    }))
    .addEdge("__start__", "echo_node")
    .addEdge("echo_node", "__end__")
    .compile();

  const output = await graph.invoke({ task: cfg.task, result: "" });
  emit({ event: "tool_call", tool: "echo_node", input: cfg.task });

  await ctx.shutdown();
  emit({ event: "done", result: output.result });
}

const cfg = loadConfig();

if (process.env.AA_SELFTEST === "1") {
  emit({ event: "started", agent_id: cfg.agentId });
  emit({ event: "tool_call", tool: "echo_node", input: cfg.task });
  emit({ event: "done", result: "selftest-ok" });
  process.exit(0);
}

await runReal(cfg);
